src/CodeGen_ARM.cpp

```diff
@@ -1647,19 +1647,26 @@ string CodeGen_ARM::mcpu_tune() const {
 }
 
 string CodeGen_ARM::mattrs() const {
+    string arch_flags;
+    string separator;
     if (target.bits == 32) {
+        if (target.has_feature(Target::ARMFp16)) {
```

Comment:
This clause could just be pulled out of the if

also, worth adding a comment that this requires an arm32 core with v8.2a extensions

Reply:
Oops, pushed the file in a half-changed state. Fixed

Reply:
The corresponding target feature says as much "Enable ARMv8.2-a half-precision floating point data processing"